File: src/database/entities/Quotations.ts
import { BaseEntity, Column, Entity, PrimaryColumn } from "typeorm";

// 종목일자별주가이력 - 호가
@Entity("ITM_DDBY_SHARPRC_HIST")
export class Quotations extends BaseEntity {
    // 거래일자
    @PrimaryColumn({
        name: "TRD_DD",
        nullable: false,
        length: 8,
    })
    private tradingDate: string;

    // 종목코드
    @PrimaryColumn({
        name: "ITM_CD",
        nullable: false,
        length: 6,
    })
    private itemCode: string;

    // 종목명
    @Column({
        name: "ITM_NM",
        length: 6,
    })
    private itemName: string;

    // 시작가격
    @Column({
        name: "STRT_PRC",
    })
    private startPrice: number;

    // 종료가격
    @Column({
        name: "END_PRC",
    })
    private endPrice: number;

    // 최고가격
    @Column({
        name: "TOP_PRICE",
    })
    private topPrice: number;

    // 최저가격
    @Column({
        name: "LWST_PRC",
    })
    private lowestPrice: number;

    // 총거래량
    @Column({
        name: "TOT_TRD_QNTY",
    })
    private totalTradingQuntity: number;

    // 공통정보
    @Column((type: any) => BaseEntity)
    private baseEntity: BaseEntity;

    public static createStockHistory = (
        tradingDate: string,
        itemCode: string,
        itemName: string,
        startPrice: number,
        endPrice: number,
        topPrice: number,
        lowestPrice: number,
        totalTradingQuntity: number,
    ): Quotations => {
        const quotation: Quotations = new Quotations();

        quotation.tradingDate = tradingDate;
        quotation.itemCode = itemCode;
        quotation.itemName = itemName;
        quotation.startPrice = startPrice;
        quotation.endPrice = endPrice;
        quotation.topPrice = topPrice;
        quotation.lowestPrice = lowestPrice;
        quotation.totalTradingQuntity = totalTradingQuntity;

        return quotation;
    };
}

File: src/main/java/com/example/kisautotrading/domain/quotations/domain/Quotations.java
package com.example.kisautotrading.domain.quotations.domain;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

// 종목일자별주가이력 - 호가
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Entity(name = "ITM_DDBY_SHARPRC_HIST")
public class Quotations {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "TRD_DD")
    private String tradingDate; // 거래일자

    @Column(name = "ITM_CD")
    private String itemCode; // 종목코드

    @Column(name = "ITM_NM")
    private String itemName; // 종목명

    @Column(name = "STRC_PRC")
    private Long startPrice; // 시가

    @Column(name = "END_PRC")
    private Long endPrice; // 종가

    @Column(name = "TOP_PRICE")
    private Long topPrice; // 최고가

    @Column(name = "LWST_PRC")
    private Long lowestPrice; // 최저가

    @Column(name = "TOT_TRD_QNTY")
    private Long totalTradingQuantity; // 총거래

    @Builder
    public Quotations(String tradingDate, String itemCode, String itemName, Long startPrice, Long endPrice, Long topPrice, Long lowestPrice, Long totalTradingQuantity) {
        this.tradingDate = tradingDate;
        this.itemCode = itemCode;
        this.itemName = itemName;
        this.startPrice = startPrice;
        this.endPrice = endPrice;
        this.topPrice = topPrice;
        this.lowestPrice = lowestPrice;
        this.totalTradingQuantity = totalTradingQuantity;
    }
}
